planner/core/stats.go

```diff
@@ -220,7 +222,30 @@ func (ts *TableScan) DeriveStats(childStats []*property.StatsInfo, selfSchema *e
 	if err != nil {
 		return nil, err
 	}
-	return ts.Source.stats, nil
+	return ts.stats, nil
+}
+
+// DeriveStats implement LogicalPlan DeriveStats interface.
+func (is *IndexScan) DeriveStats(childStats []*property.StatsInfo, selfSchema *expression.Schema, childSchema []*expression.Schema) (*property.StatsInfo, error) {
+	for i, expr := range is.AccessConds {
+		is.AccessConds[i] = expression.PushDownNot(is.ctx, expr)
+	}
+	is.stats = is.Source.deriveStatsByFilter(is.AccessConds)
+	if len(is.AccessConds) == 0 {
+		is.Ranges = ranger.FullRange()
+	}
+	// TODO: If the AccessConds is not empty, we have set the range when push down the selection.
```

Comment:
This will bring too much codes?

Reply:
This requires the rule `PushSelDownIndexScan`.  And I prefer to implement it in the next PR.